### PR TITLE
New debt coupon UI

### PR DESCRIPTION
**Pull request by chriswudev**

### CODE REVIEW

**Comment by Zequez** [comment]
Looking good so far! Keep it up! Thanks!

**Review by Zequez — COMMENTED** [review]
We'll merge this when the whole #38 is finished;
@chriswudev remember to tag the issues on the PR description as you check them off, so they get linked back to this PR; and feel free to edit the issues and check the boxes!

File: components/common/contracts-shortcuts.ts [hunk 1]
```diff
@@ -30,6 +32,8 @@ export async function accountBalances(account: EthAccount, contracts: Contracts)
     erc1155BalanceOf(account.address, contracts.debtCouponToken),
     erc1155BalanceOf(account.address, (contracts.bondingToken as unknown) as ERC1155Ubiquity),
     contracts.usdc.balanceOf(account.address),
+    erc1155BalanceOf(account.address, (contracts.uBOND as unknown) as ERC1155Ubiquity), // uBOND
```
Comment (Zequez) [inline components/common/contracts-shortcuts.ts:35]: uBOND will be like uAR, it's an ERC20 token, so a simple balanceOf should suffice.

File: contracts.ts [hunk 2]
```diff
@@ -213,6 +215,8 @@ export async function connectedContracts(): Promise<{
       crvToken: contracts.crvToken(addr.crvToken, provider),
       bondingToken: contracts.bondingToken(addr.bondingToken, provider),
       debtCouponToken: contracts.debtCouponToken(addr.debtCouponToken, provider),
+      uBOND: contracts.bondingToken(addr.bondingToken, provider),
```
Comment (Zequez) [inline contracts.ts:234]: Confusingly, uBOND doesn't stand for the bondingToken 😆 
For now just put the uAR contract here, as the uBOND token doesn't exist yet.

File: contracts.ts [hunk 3]
```diff
@@ -213,6 +215,8 @@ export async function connectedContracts(): Promise<{
...
+      uDEBT: contracts.debtCouponToken(addr.debtCouponToken, provider),
```
Comment (Zequez) [inline contracts.ts:235]: We already have the debtCouponToken, why duplicate it?

File: components/common/contracts-shortcuts.ts [hunk 4]
```diff
@@ -30,6 +32,8 @@ export async function accountBalances(account: EthAccount, contracts: Contracts)
...
+    erc1155BalanceOf(account.address, contracts.uDEBT), // uDEBT
```
Comment (Zequez) [inline components/common/contracts-shortcuts.ts:36]: This returns the total balance as a BigNumber sum. But for the UI we'll need each token differentiated; check how the `erc1155BalanceOf` is created, and you'll see we first get a list of IDs for an address, and then we call the balanceOf of each token ID. We need the ID and the balance both, so we can then use it to sell these tokens individually.
Then we'll also need to expiration block, but we'll add a pin to that as we need to check how to get it from the DebtCoupon contract.

File: components/DebtCoupon.tsx [hunk 5]
```diff
@@ -1,222 +1,521 @@
...
+  const udebtUbqRedemptionRate = 0.25;
+  const uadTotalSupply = 233000;
+  const ubondTotalSupply = 10000;
+  const uarTotalSupply = 30000;
+  const udebtTotalSupply = 12000;
+  const coupons: Coupons = {
+    uDEBT: [
+      { amount: 1000, expiration: 1640390400000, swap: { amount: 800, unit: "uAR" } },
+      { amount: 500, expiration: 1639526400000, swap: { amount: 125, unit: "uAR" } },
+      { amount: 666, expiration: 1636934400000, swap: { amount: 166.5, unit: "UBQ" } },
```
Comment (Zequez) [inline components/DebtCoupon.tsx:139]: Good idea the swap object for the mock!

File: contracts.ts (hunk 2 above)
Comment (0x4007) [reply contracts.ts:234]: Open to ideas for renaming the tokens (we have a rename function!)

**Comment by 0x4007** [comment]
@marcomow @Zequez should this be merged? Can you guys merge it if so?

**Comment by sergfeldman** [comment]
Chris:
This pull request was implemented in another branch. 
The Master branch already has all the changes implemented in this pull request. 
That`s why there was no need to keep this pull request open.

**Pull request closed without merging**